### ISSUE TITLE
YAML External File Variable Resolution Issue

### ISSUE THREAD

**Issue by joshwils82** [serverless/serverless#2522]
# This is a Bug Report

## Description

Including a yaml variables file without a property defined causes issues when used to override variables.  I have 2 yaml files I am using to resolve variables.  A common.yml and a stage.yml.  If the stage.yml is missing the property, then when serverless attempts to resolve variables using multiple sources, the variables are resolved to empty object instead of the expected value.

For bug reports:
* What went wrong?
The variables are not resolved correctly and instead are resolved to an empty object

* What did you expect should have happened?
The yaml file without the variable should have no affect on variable resolution and the variables should be resolved from common.yml correctly.

* What was the config you used?

Example common.yml

```
test:
  bug: testing the bug
```

Example stage.yml

```
placeholder: placeholder
```

Example Variables
```
custom:
  vars:
    common: ${file(./variables/common.yml)}
    stage: ${file(./variables/${self:provider.stage}.yml)}
  test: ${self:custom.vars.stage.test.bug, self:custom.vars.common.test.bug}
```

* What stacktrace or error message from your provider did you see?

No stacktrace.  Debugging shows the following value for the variable:

```
{
  "vars": {
    "common": {
      "test": {
        "bug": "testing the bug"
      }
    },
    "stage": {
      "placeholder": "placeholder"
    }
  },
  "test": {}
}
```

I expected the test property to contain "testing the bug"

Similar or dependent issues:

## Additional Data

* ***Serverless Framework Version you're using***: 1.0.3
* ***Operating System***: Linux
* ***Stack Trace***:
* ***Provider Error messages***:

**Comment by miltador** [comment]
I think I've got a fix for this and some other related issues. Will test it out and create a PR.

**Comment by eahefnawy** [comment]
closing this issue as it should now be fixed with #2541 ... @joshwils82 feel free to open if you're still facing issues with this 😊
